src/network-mgr.cpp

```diff
@@ -2,12 +2,99 @@
 #include <QNetworkAccessManager>
 #include <QNetworkProxy>
 #include <algorithm>
+#include <QStringRef>
+#include <QSslConfiguration>
+#include <QSslSocket>
+#include <QSslCipher>
 namespace {
 QNetworkProxy proxy_;
+const char *const kWhitelistCiphers[] = {"ECDHE-RSA-AES256-GCM-SHA384"
+                                         "ECDHE-RSA-AES128-GCM-SHA256"
+                                         "DHE-RSA-AES256-GCM-SHA384"
+                                         "DHE-RSA-AES128-GCM-SHA256"
+                                         "ECDHE-RSA-AES256-SHA384"
+                                         "ECDHE-RSA-AES128-SHA256"
+                                         "ECDHE-RSA-AES256-SHA"
+                                         "ECDHE-RSA-AES128-SHA"
+                                         "DHE-RSA-AES256-SHA256"
+                                         "DHE-RSA-AES128-SHA256"
+                                         "DHE-RSA-AES256-SHA"
+                                         "DHE-RSA-AES128-SHA"
+                                         "ECDHE-RSA-DES-CBC3-SHA"
+                                         "EDH-RSA-DES-CBC3-SHA"
+                                         "AES256-GCM-SHA384"
+                                         "AES128-GCM-SHA256"
+                                         "AES256-SHA256"
+                                         "AES128-SHA256"
+                                         "AES256-SHA"
+                                         "AES128-SHA"
+                                         "DES-CBC3-SHA"};
+// return false if it contains RC4, RSK, CBC, MD5, DES, DSS, EXPORT, NULL
+bool isWeakCipher(const QString& cipher_name)
+{
+    int current_begin = 0;
+    int current_end;
+    QStringRef name;
+    while((current_end = cipher_name.indexOf("-", current_begin)) != -1) {
+        name = QStringRef(&cipher_name, current_begin, current_end - current_begin);
+        if (name == "RC4")
+            return true;
+        else if (name == "PSK")
+            return true;
+        else if (name == "CBC")
+            return true;
+        else if (name == "MD5")
+            return true;
+        else if (name == "DES")
+            return true;
+        else if (name == "DSS")
+            return true;
+        else if (name == "EXP")
+            return true;
+        else if (name == "NULL")
+            return true;
```

Comment:
@lins05 sorry, but I can't agree with you about using `QSet` and `QHashMap` everywhere. It just makes no sense to use `QSet` here but wastes a lot of CPU time if we can use plain `vector` instead. (Internally, the methods of `QSet` and `QHashMap` call hundred times of instructions than plain `vector` do, cost much more memory and don't scale well)